clippy_lints/src/from_over_into.rs

```diff
@@ -80,6 +80,11 @@ impl<'tcx> LateLintPass<'tcx> for FromOverInto {
             && cx.tcx.is_diagnostic_item(sym::Into, middle_trait_ref.def_id)
             && !matches!(middle_trait_ref.substs.type_at(1).kind(), ty::Alias(ty::Opaque, _))
         {
+            if !target_ty.find_self_aliases().is_empty() {
+                // It's tricky to expand self-aliases correctly, we'll ignore it to not cause a
+                // bad suggestion/fix.
+                return;
+            }
```

Comment:
It can go in `convert_to_from` so that we still lint these cases, just that instead of a full suggestion it'll ask you to figure it out yourself

The test case will probably need to be moved to `from_over_into_unfixable.rs` (or however it's spelled) also

Reply:
Ah I see, thanks a lot for the patience, now it produces a lint without a suggestion!